### PR TITLE
feat(enhancement): Allow escorts to cloak when damaged

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Feature:** This PR implements the feature request detailed and discussed in issue #2916

## Feature Details
Player-owned escorts will try to cloak and escape the battle to repair when considerably damaged if there are enemies nearby (it's the same as for non-player ships).

## Testing Done
My Shield Beetle escort cloaked when its shields dropped to zero and its hull was about 100%. The exact moment probably depends on the shields / hull ratio of the ship model.

## Performance Impact
N/A

### CODE REVIEW

**Comment by Amazinite** [comment]
I feel like this should be activated by default. It's probably more often the case that people would prefer that their escorts use cloaks when they need them as opposed to needing to manually track whether to cloak the entire fleet for the sake of one ship (assuming you have a fleet with multiple cloaking ships).

**Comment by vitalchip** [comment]
I think it is also worth adding a !IsYours() check to cloakfreely otherwise player ships with sufficient fuel generation will cloak at will and the player won't be able to uncloak them / select them etc until an enemy appears. It's not a useful mechanic for a player and more likely to be frustrating.

File: source/AI.cpp [hunk 1]
```diff
@@ -628,7 +628,7 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 				command |= Command::CLOAK;
 		}
 		// Cloak if the AI considers it appropriate.
-		else if(DoCloak(*it, command))
+		if((!it->IsYours() || (!isCloaking && Preferences::Has("Defensive escort cloaking"))) && DoCloak(*it, command))
```
Comment (warp-core) [inline source/AI.cpp:631]: Split this into multiple if statements for readability.

**Comment by quyykk** [comment]
Do we really need a preference for this? At least to me it looks like something one would always enable.

**Comment by quyykk** [comment]
Is there a scenario where one would disable the preference?

**Comment by Saugia** [comment]
I have to agree with Quyykk here. I don't see the reasoning for making this a preference, rather, if this is a behavior desired. have it incorporated into how the AI uses cloaking. Though, that then begs the question of what we deem proper cloaking Ai behavior. In my opinion, if an escort has a cloak, it would automatically cloak if has taken significant damage. 

I'm not sure if that should be 0% shields though, as there are some ships in the game that are hull-tanks and not shield-tanks. I'd lean towards at 60% hull or something.

Regardless - we don't need a preference for every little thing, and we shouldn't be adding a bunch of preferences. That's my 2 cents.

**Comment by lumbar527** [comment]
Yeah, I guess a preference doesn't make sense. And now I see that it's not just you, it's your pilots. They might have a sense of self-preservation!

**Comment by quyykk** [comment]
> I'm not sure if that should be 0% shields though, as there are some ships in the game that are hull-tanks and not shield-tanks. I'd lean towards at 60% hull or something

There's a ship health value we can use :smile:  https://github.com/endless-sky/endless-sky/blob/master/source/Ship.cpp#L3327

**Review by quyykk — APPROVED** [review]

**Pull request merged**